pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -1188,3 +1209,71 @@ func (s *discovery) getServices(params params) (*model.ServicesResponse, error)
 
 	return response, nil
 }
+
+// getServices processes a list of PIDs and returns service information for each.
+// This is used by the /services endpoint which accepts explicit PID lists and bypasses
+// the port retry logic used by the /check endpoint.
```

Comment:
_Why_ is it bypassing the port retry logic? I guess it's because it will be done in the core agent? Maybe that should be mentioned here.